sphinx/ext/autodoc/__init__.py

```diff
@@ -719,6 +727,21 @@ def is_filtered_inherited_member(name: str, obj: Any) -> bool:
                     doc = obj.docstring
 
                 doc, metadata = separate_metadata(doc)
+
+                # workaround for `TypedDict` subclassing; does not work with
+                # `typing_extensions.TypedDict` and requires explicitly passing
+                # `TypedDict` to subclasses
+                if not doc and is_typeddict(self.object):
+                    __annotations__ = self.get_attr(self.object, '__annotations__', {})
+                    if membername in __annotations__:
+                        original_bases = safe_getattr(self.object, '__orig_bases__', ())
```

Comment:
In Python 3.12+, `__orig_bases__` can be retrieved via `types.get_original_bases` but I'm not sure if it's sufficient.